n3fit/src/n3fit/layers/rotations.py

```diff
@@ -91,6 +91,28 @@ def call(self, pdf_raw):
         # Concatenating destroys the batch index so we have to regenerate it
         return op.batchit(ret)
 
+class AddPhoton(MetaLayer):
+    """
+    Changes the value of the photon to non-zero.
+
+    Both input and output have shape (1, None, 14). However,
+    the input has a photon which is set to zero, while for the
+    output it is set to its real value.
+    """
+
+    def __init__(self, output_dim=14, **kwargs):
+        self.output_dim = output_dim
+        super().__init__(**kwargs)
+    
+    def call(self, pdfs, pdf_ph):
```

Comment:
Is the photon changed at every epoch of the NN?

Reply:
Definitely not, it is changed at every iteration (and so most of the time once per fit)

Reply:
Ok, then this layer needs to change. @niclaurenti the value of `pdf_ph` should enter as the input of the class. In any way it enters you can then reshape it to the right shape (in practice `(replicas, basis_size, xgrid)`) and below the layer should only return `op.stack([pdfs, self.pdf_ph], axis=1)`

I don't remember if the `basis_size` is `axis=1`.

Reply:
This would be fine, is what I suggested in the first place, but unfortunately is not possible: when you create an instance (in this file) the photon is not available, since the x values are not yet available.

That's why I proposed to move it to an input of the NN.

Reply:
You can always do `op.stack([pdfs[1:], self.pdf_ph], axis=1)` to skip the photon.

Or we can modify the rotation to be just 13 elements (instead of 14) and add always the photon, sometimes as 0s sometimes as an actual photon.

Reply:
Indeed, this is what I suggested here https://github.com/NNPDF/nnpdf/pull/1643#discussion_r1038395552 (even though you have to swap to put the photon first, but that's fine).
We just need to make the photon available here.